### DISCUSSION TITLE
Location XML tags specifying a track's color and line width before and after editing appearance

### DISCUSSION THREAD

**Discussion by tmusolf** [osmandapp/OsmAnd#17015] [Q&A]
When importing a GPX file containing multiple tracks you will be given the option to have OSMAnd break the gpx file up into individual track files.  This will happen if the track count is between 2-50 tracks in the original gpx file.

These new track files contain tags describing the track's line width and color.  
```
    <extensions>
      <osmand:color>#550F9D58</osmand:color>
      <osmand:width>1.0</osmand:width>
    </extensions>
```
Strangely, theses tags are in two places in the gpx file.  One is within the <trk> tag and the other is outside of the <trk> tag, within the <gpx> tag.  Doing some hand editing of the files it appears that OSMAnd uses the values within the <gpx> tag for the initial display of the track.

You can use the OSMAnd edit appearance option to change the line width and color.  However, these new width and color values are not written to the gpx file.   

Shutting down and restarting OSMAnd retains these modified width/color values. 

If I use the OSMAnd My places delete track option to remove the tracks and then reimport them then the track's original width/color values are restored.

If however I delete the track with a file utility and then reimport it the values modified before the delete are retained.

This indicates that OSMAnd storing these values modified width/color values in some other location and not the track gpx file.  Does anyone know where this information is being stored and why the values modified using edit appearance are not written to the gpx file?

**Comment by colorimeter** [comment]
If you changes the style of a track in the App, it stores the values in its own gpx database, but you can style them by modifying the gpx itself by inserting an extensions tag inside each <trk> tag: (AArrggbb):
```
  <trk>
    <extensions>
     <color>#ff00ff00</color>
    </extensions>
    <trkseg>

    ....

```

Or with the [well documented](https://osmand.net/docs/technical/osmand-file-formats/osmand-gpx/) osmand style extension just inside the gpx tag:
```
<gpx ...>
  <extensions>
    <osmand:color>#4e4eff</osmand:color>
    <osmand:width>bold</osmand:width>
    <osmand:show_arrows>true</osmand:show_arrows>
    <osmand:split_type>distance</osmand:split_type>
    <osmand:split_interval>2000.0</osmand:split_interval>
  </extensions>
</gpx>
```

- The gpx database has higher priority than the file itself so you have to reset the setting to default, if you have already changed in the app. 
- Changing the already displayed gpx needs app restart to be redrawn with the new style.
